addons/l10n_ke_edi_tremol/models/account_move_send.py

```diff
@@ -0,0 +1,43 @@
+from odoo import _, models
+
+class AccountMoveSend(models.Model):
+    _inherit = 'account.move.send'
+
+    def action_send_and_print(self, from_cron=False, allow_fallback_pdf=False):
+        # Override to allow generating the proforma pdf when the invoice has not been
+        # sent to the fiscal device yet
+        self.ensure_one()
+        if not self.move_ids.filtered(lambda m: m.country_code != 'KE'):
+            # all Kenyan invoices
+            allow_fallback_pdf = True
+        super().action_send_and_print(from_cron, allow_fallback_pdf)
+
+    def _hook_invoice_document_before_pdf_report_render(self, invoice, invoice_data):
+        super()._hook_invoice_document_before_pdf_report_render(invoice, invoice_data)
+        if invoice.country_code == 'KE' and (
+            not invoice.l10n_ke_cu_invoice_number \
+            or not invoice.l10n_ke_cu_serial_number \
+            or not invoice.l10n_ke_cu_qrcode \
+            or not invoice.l10n_ke_cu_datetime
+        ):
+            invoice_data['error'] = _(
+                "The document does not have details related to the fiscal device, a proforma invoice will be sent."
+            )
+
+    def _compute_send_mail_extra_fields(self):
+        super()._compute_send_mail_extra_fields()
+        warning_msg = ""
+        for wizard in self:
+            for invoice in self.move_ids:
+                if invoice.country_code == 'KE' and (
+                    not invoice.l10n_ke_cu_invoice_number \
+                    or not invoice.l10n_ke_cu_serial_number \
+                    or not invoice.l10n_ke_cu_qrcode \
+                    or not invoice.l10n_ke_cu_datetime
+                ):
+                    warning_msg += f"{invoice.name} : " + _(
+                        "The document does not have details related to the fiscal device, a proforma invoice will be sent.") + "\n"
+            if wizard.send_mail_warning_message:
```

Comment:
I have some doubt on `send_mail_warning_message` because that relates to the Email checkbox and email sending process, which is not the Kenyan flow I guess. 

Here is a reference to my WIP for Italy... I'm not certain this is the way to go for Kenya but just to show you some context: https://github.com/odoo/odoo/pull/142596/files#diff-a93d4442f33dead3e87d7e0e4ef601e1aa14bd48d6810555cf3354a5003c3fe1L10

Thing is, every EDI that needs sending expands this wizard... PEPPOL, Italy, Kenya, I guess more - we can't all just use a single warning field.
Adding another warning computed field would require showing it in a view - this would require the user to upgrade the module.